cmd/livepeer/livepeer.go

```diff
@@ -183,14 +171,68 @@ func main() {
 	}
 
 	var gethCmd *exec.Cmd
-	//Set up ethereum-related stuff
-	if *ethDatadir != "" && !*offchain {
-		gethipc := filepath.Join(filepath.Join(*ethDatadir, "geth.ipc"))
-		//If getipc file is not there, start the geth node
-		if _, err := os.Stat(gethipc); os.IsNotExist(err) {
-			//Set up geth params depending on network, invoke geth
+	if *offchain {
+		glog.Infof("***Livepeer is in off-chain mode***")
+	} else {
+		var acct accounts.Account
+		var keystoreDir string
+		if _, err := os.Stat(*ethKeyPath); !os.IsNotExist(err) {
+			//Try loading eth key from ethKeyPath
+			data, err := ioutil.ReadFile(*ethKeyPath)
+			if err != nil {
+				glog.Errorf("Cannot read key from %v", *ethKeyPath)
+				return
+			}
+
+			var objmap map[string]*json.RawMessage
+			if err := json.Unmarshal(data, &objmap); err != nil {
+				glog.Errorf("Cannot parse key from %v", *ethKeyPath)
+				return
+			}
+			var addr string
+			if err := json.Unmarshal(*objmap["address"], &addr); err != nil {
+				glog.Errorf("Cannot find address in %v", *ethKeyPath)
+				return
+			}
+			keystoreDir, _ = filepath.Split(*ethKeyPath)
+			acct, err = getEthAccount(keystoreDir, addr)
+			if err != nil {
+				glog.Errorf("Cannot get account %v in %v", addr, keystoreDir)
+				return
+			}
+		} else {
+			//Try loading eth key from datadir
+			keystoreDir = filepath.Join(*datadir, "keystore")
+			if _, err := os.Stat(keystoreDir); !os.IsNotExist(err) {
+				acct, err = getEthAccount(keystoreDir, *ethAcctAddr)
+				if err != nil {
+					glog.Errorf("Cannot get account %v from %v", *ethAcctAddr, *datadir)
+					return
+				}
+			}
+		}
+
+		if acct.Address.Hex() == "" {
+			glog.Errorf("Cannot find eth account")
+			return
+		}
+		if keystoreDir == "" {
+			glog.Errorf("Cannot find keystore directory")
+			return
+		}
+
+		//Get the Eth client connection information
+		gethUrl := ""
+		if *ethIpcPath != "" {
+			//Connect to specified IPC file
+			gethUrl = *ethIpcPath
+		} else if *ethWsUrl != "" {
+			//Connect to specified websocket
+			gethUrl = *ethWsUrl
+		} else {
+			//Default behavior - start a Geth node and connect to its IPC
 			if *testnet {
-				gethCmd = exec.Command("geth", "--rpc", "--datadir", *ethDatadir, "--networkid=858585", "--bootnodes=enode://a1bd18a737acef008f94857654cfb2470124d1dc826b6248cea0331a7ca82b36d2389566e3aa0a1bc9a5c3c34a61f47601a6cff5279d829fcc60cb632ee88bad@13.58.149.151:30303") //timeout in 3 mins
+				gethCmd = exec.Command("geth", "--rpc", "--datadir", defaultEthDatadir, "--networkid=858585", "--bootnodes=enode://a1bd18a737acef008f94857654cfb2470124d1dc826b6248cea0331a7ca82b36d2389566e3aa0a1bc9a5c3c34a61f47601a6cff5279d829fcc60cb632ee88bad@13.58.149.151:30303") //timeout in 3 mins
```

Comment:
It could be nice to read these default values from a config. Could be tracked in a separate issue.